src/QueryDataTable.php

```diff
@@ -174,34 +174,40 @@ public function prepareCountQuery(): QueryBuilder
 
         if ($this->isComplexQuery($builder)) {
             $builder->select(DB::raw('1 as dt_row_count'));
-            if ($this->ignoreSelectInCountQuery || ! $this->isComplexQuery($builder)) {
-                return $this->getConnection()
-                    ->query()
-                    ->fromRaw('('.$builder->toSql().') count_row_table')
-                    ->setBindings($builder->getBindings());
+            $clone = $builder->clone();
+            $clone->setBindings([]);
+            if ($clone instanceof EloquentBuilder) {
+                $clone->getQuery()->wheres = [];
+            } else {
+                $clone->wheres = [];
             }
 
-            $builder = clone $this->query;
+            if ($this->isComplexQuery($clone)) {
+                if (!$this->ignoreSelectInCountQuery) {
+                    $builder = clone $this->query;
+                }
 
-            return $this->getConnection()
-                ->query()
-                ->fromRaw('('.$builder->toSql().') count_row_table')
-                ->setBindings($builder->getBindings());
+                return $this->getConnection()
+                            ->query()
+                            ->fromRaw('(' . $builder->toSql() . ') count_row_table')
+                            ->setBindings($builder->getBindings());
+            }
         }
-
         $row_count = $this->wrap('row_count');
         $builder->select($this->getConnection()->raw("'1' as {$row_count}"));
```

Comment:
Maybe we can consider @abdulrhmansouda suggestion of removing the orders?

```suggestion
        $builder->select($this->getConnection()->raw("'1' as {$row_count}"))->reorder();
```